Letterbook.Adapter.ActivityPub/Types/ProfileActor.cs

```diff
@@ -0,0 +1,174 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using ActivityPub.Types;
+using ActivityPub.Types.AS;
+
+namespace Letterbook.Adapter.ActivityPub.Types;
+
+/// <summary>
+/// A unified actor type that implements all the actor extensions Letterbook supports
+/// </summary>
+/// <remarks>The derived types support type checking and mapping to specific AP Actor types.</remarks>
+public class ProfileActor : APActor, IASModel<ProfileActor, ProfileActorEntity>
+{
+	private ProfileActorEntity Entity { get; }
+
+	public ProfileActor() => Entity = TypeMap.Extend<ProfileActor, ProfileActorEntity>();
+
+	public ProfileActor(TypeMap typeMap, bool isExtending = true) : base(typeMap, false)
+		=> Entity = TypeMap.ProjectTo<ProfileActor, ProfileActorEntity>(isExtending);
+
+	public ProfileActor(ASType existingGraph) : this(existingGraph.TypeMap)
+	{
+	}
+
+	[SetsRequiredMembers]
+	public ProfileActor(TypeMap typeMap, ProfileActorEntity? entity) : base(typeMap, null)
+		=> Entity = entity ?? typeMap.AsEntity<ProfileActor, ProfileActorEntity>();
+
+	static ProfileActor IASModel<ProfileActor>.FromGraph(TypeMap typeMap) => new(typeMap, null);
+
+	public ASLink? SharedInbox
+	{
+		get => Entity.SharedInbox;
+		set => Entity.SharedInbox = value;
+	}
+
+	public List<PublicKey> PublicKeys
+	{
+		get => Entity.PublicKey;
+		set => Entity.PublicKey = value;
+	}
+}
+
+public sealed class ProfileActorEntity : ASEntity<ProfileActor, ProfileActorEntity>
+{
+	[JsonPropertyName("publicKey")] public List<PublicKey> PublicKey { get; set; } = new();
+
+	[JsonPropertyName("sharedInbox")] public ASLink? SharedInbox { get; set; }
+}
+
+public class ProfilePersonActor : ProfileActor, IASModel<ProfilePersonActor, ProfilePersonActorEntity>
```

Comment:
This setup seems to be working well, except that it serializes with `"type": ["Person", "Object"]`

full example:
```json
{
  "type": [
    "Object",
    "Person"
  ],
  "@context": "https://www.w3.org/ns/activitystreams",
  "id": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt",
  "name": "admin",
  "published": "2024-10-08T00:00:00",
  "summary": "",
  "updated": "2024-10-08T02:38:01.897342Z",
  "inbox": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt/inbox",
  "outbox": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt/outbox",
  "following": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt/following",
  "followers": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt/followers",
  "preferredUsername": "admin",
  "publicKey": [
    {
      "id": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt#key-0",
      "publicKeyPem": "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA2655NZbkI8aZpPUkTFiU\ni5ZFerSWQZoeOj/G/4ElYve181vbatdgvupW76KQIWU4FMq4a6WOH6Rr+HLC3Kxr\no1ahIAueVxkW4ynWQVbyKZdtE1FjlX3VLHFzNW3k1rwvQeBfBAYhXx//HBVcz2Qs\nEhc5LE9psT7HU0hlLRE+FRfV2f9Z2ct3oLOwyt0Jxnp753GUKIlzKSj6DlRuDy+D\n1wVwJ2fwD9/tVRWLx3QRGavXqbPCvGUN7VlWYtIYWzF4hOBZrxtTC/21h8TWf4NK\nGB2m7fvN+QfsdIl+LhjXj0I5abJeuT1eKRnNv3FFz9Pt+glGG13mh9Jrp0B3if0y\n7wIDAQAB\n-----END PUBLIC KEY-----"
    },
    {
      "id": "https://host.castle/actor/06e56srrtx76k3ut95y0swkxt#key-1",
      "publicKeyPem": "-----BEGIN PUBLIC KEY-----\nMIGbMBAGByqGSM49AgEGBSuBBAAjA4GGAAQAJIG4Jz+9UdpaIQ8XQyEBX9h6cR/C\nNoBkGeDpD0jsTEODZpSVH0FsM8wYlDsaCBQ+EEF7lbfl9Fr5SR/BTR/1x7wAweTv\njsfcvxdLoFuYcPfNLxpM+7wNkIRJsg41TcZj84KYH+gNggsLOeymafDGwQR5Ci72\nXHLkU+mJHGBQbInG2Wo=\n-----END PUBLIC KEY-----"
    }
  ],
  "sharedInbox": "https://host.castle/actor/shared_inbox"
}
```